### ISSUE TITLE
Enhance max-statements-per-line to allow simple arrow functions

### ISSUE THREAD

**Issue by TWiStErRob** [eslint/eslint#9210]
**Tell us about your environment**

* **ESLint Version:** 4.6.0
* **Node Version:** 5.6.0
* **npm Version:** 3.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** default, I guess

**Please show your full configuration:**

```js
{
	"env": {
		"es6": true,
	},
	"rules": {
		"max-statements-per-line": ["error", { "max": 1 }]
	}
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
var input = [];

if (input) { input.push({}); } // line 3: should fail
if (input) { input.push({}); } else { input = []; } // line 4: should fail

input.forEach(item => { item.prop = "..." }); // line 6: should pass

var arrow = item => { item.prop = "..." }; // line 8: should pass
```

**What did you expect to happen?**
See inline comments.

I would like to have an **option** to allow simple functions/arrows to be excluded from the statement count for single-invocation outer statements. This came up because I have code that is doing some initialization like line 6 in a `_.chain`.

**What actually happened? Please include the actual, raw output from ESLint.**
```
  3:14  error  This line has 2 statements. Maximum allowed is 1  max-statements-per-line
  4:14  error  This line has 3 statements. Maximum allowed is 1  max-statements-per-line
  6:25  error  This line has 2 statements. Maximum allowed is 1  max-statements-per-line
  8:23  error  This line has 2 statements. Maximum allowed is 1  max-statements-per-line
```

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
